src/keyconfig.rs

```diff
@@ -218,8 +194,13 @@ impl KeyConfig {
       &self.context_menu,
       &self.next_tab,
       &self.previous_tab,
-    ];
+    ]
+  }
+
+  pub fn check_duplicates(&self, mut elements: Vec<&KeyCode>) -> Result<()> {
     let l = elements.len();
+    // TODO: Write Ord implementation for KeyCode.
+    // Vecs need to be sorted for dedup to work correctly.
```

Comment:
Can you elaborate on why they need to be sorted for dedup to work correctly?

Reply:
Sure! 

I think the docs explain it better than we could(see example, also copied from the docs).
https://doc.rust-lang.org/src/alloc/vec/mod.rs.html#2673
> Removes consecutive repeated elements in the vector according to the [PartialEq](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html) trait implementation.
> 
> If the vector is sorted, this removes all duplicates.
> Examples
```
let mut vec = vec![1, 2, 2, 3, 2];
vec.dedup();
assert_eq!(vec, [1, 2, 3, 2]);
```

Due to the Vector not being sorted conflicting elements won't necessarily be consecutive.

That's what triggered the duplicate key error of #579: 
edit and duplicate were set to the same value in update and by chance follow each other in the duplicate key check.

Reply:
Oh my, I had no idea that `.dedup` needed to be sorted! That's embarrassing. 

Do we need to implement `Ord` or can be get by by using `.sort_by` with a custom comparator?

Reply:
We only stumbled about `.dedup` needing a sorted array because a friend told it to us a year ago.
Having a `.sort_then_dedup` would be a great clarity feature, as we otherwise probably also have made that mistake ^^'.

Assuming that all `KeyCode`s are of Variant `Char`, we could use `char`s `Ord` to do it for us.
If the `KeyCode`s can be of other Variants too then we'd probably need to manually implement `Ord` and may have to rethink key handling in general(we e.g. have no idea how we would compare CtrlAlt Variant to b)